mithril-common/src/cardano_transaction_parser.rs

```diff
@@ -119,43 +116,54 @@ impl CardanoTransactionParser {
 
     /// Read blocks from immutable file
     fn read_blocks_from_immutable_file(immutable_file: &ImmutableFile) -> StdResult<Vec<Block>> {
-        let cbor = fs::read(&immutable_file.path).with_context(|| {
-            format!(
-                "CardanoTransactionParser could not read from immutable file: {:?}",
+        let cardano_blocks_reader =
+            CardanoTransactionParser::cardano_blocks_reader(immutable_file)?;
+
+        let mut blocks = Vec::new();
+        for block in cardano_blocks_reader {
+            let block = block.map_err(|e| {
+                anyhow!(e).context(format!(
+                    "Error while reading block in immutable file: '{:?}'",
+                    immutable_file.path
+                ))
+            })?;
+            blocks.push(CardanoTransactionParser::convert_to_block(
+                &block,
+                immutable_file,
+            )?);
+        }
+        Ok(blocks)
+    }
+
+    fn convert_to_block(block: &[u8], immutable_file: &ImmutableFile) -> StdResult<Block> {
+        let multi_era_block = MultiEraBlock::decode(block).map_err(|e| {
```

Comment:
Same comment as previously, prefer use of implicit conversion using `with_context`.

Reply:
Done